configure

```diff
@@ -1545,7 +1545,7 @@ Optional Packages:
   --with-trace-refs       enable tracing references for debugging purpose
                           (default is no)
   --with-assertions       build with C assertions enabled (default is no)
-  --with-lto              enable Link-Time-Optimization in any build (default
+  --with-lto[=no|thin]    enable Link-Time-Optimization in any build (default
                           is no)
```

Comment:
the current `no` means that whether we apply LTO optimization or not.
So the help guide should be updated if we also supply the thin LTO optimization pass.

The following options should be available.
A. no lto (by default)
B. with lto (default=full)
C. with lto (full designated)
D. with lto (thin lto designated)